### PR TITLE
[315] Create New Tank

### LINKED ISSUE #315: As a user, I can create a new Tank
Acceptance Criteria

- [x] A new route is added that maps to Tank#new
- [x] A new view is added to this route that has a form for creating a new Tank record
- [x] This form can be submitted to add a new Tank record to the database
- [x] A feature test is added that loads this form and submits it, creating a new record.

There is some prior art for how we want new record form in our application to look and behave. See Facilities#new and Animals#new. If this task is picked up before #313,  Please add to the scope of that issue, or create a new issue for...

- [x] A link to this form is added to the `Tanks#index` page. 

The existing pattern in our application for `new` forms is to redirect to the records `show` page with a success message after the new record is created. If #316 is not completed yet when this task is picked up, you may redirect to the index page, or re-render the new form with a success message, and follow up with the correct behavior once #316 is completed

### PR DESCRIPTION
**Pull request by darrendc**
Resolves #315

### Description
This allows a user to create a new Tank via a `new` tank form

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

A user can view the form properly. The form can be filled out and see that it was successfully created on the show page.

### Screenshots
Screen Shot 2020-09-21 at 5 12 01 PM

### CODE REVIEW

**Review by benreyn — APPROVED** [review]
Some thoughts about the `for_organization` scope added here, but consider them non-blocking unless you have the appetite to make changes.

File: app/models/facility.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@ class Facility < ApplicationRecord
   belongs_to :organization
   after_commit { Rails.cache.delete('facility_codes') }
 
+  scope :for_organization, ->(organization_id) { where organization_id: organization_id }
```
[the remark this answers is not in the document]
Comment (darrendc) [reply app/models/facility.rb]: I agree. Made this chane. @haydenrou is working on this model in https://github.com/rubyforgood/abalone/pull/351 so tests can built around his setup

File: app/models/tank.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ class Tank < ApplicationRecord
 
   has_one :family, required: false
 
-  delegate :name, to: :facility, prefix: true
+  delegate :name, to: :facility, prefix: true, allow_nil: true
```
Comment (smkopp92) [inline app/models/tank.rb:11]: Nice update here!

File: app/views/tanks/index.html.erb [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <div class="container facilities__index__container">
     <div class="facilities__index__container__header">
       <h1 class="title">Tanks</h1>
-      <%= link_to "#", class: "button-primary" do %>
+      <%= link_to new_tank_path, class: "button-primary" do %>
```
Comment (smkopp92) [inline app/views/tanks/index.html.erb:5]: 🙌

File: app/views/tanks/show.html.erb [hunk 4]
```diff
@@ -1,6 +1,5 @@
 <section class="section facilities__show">
   <div class="container facilities__show__container">
-    <p id="notice"><%= notice %></p>
```
Comment (smkopp92) [inline app/views/tanks/show.html.erb:3]: Good catch here

File: app/models/facility.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (haydenrou) [reply app/models/facility.rb]: I'll make the change there too, good idea!

**Pull request merged**